### DISCUSSION TITLE
Can Cosmos simulate millions or even tens of millions of vertices under poor performance? What GPU and memory capacity are needed?

### DISCUSSION THREAD

**Discussion by XM-8JD2** [cosmograph-org/cosmos#109] [Q&A] [answered]
Can Cosmos simulate millions or even tens of millions of vertices under poor performance? What GPU and memory capacity are needed?

**Comment by rokotyan** [comment, marked as answer]
Hi @XM-8JD2, I would say you can go up to a million, maybe a couple millions max. There are some WebGL limitations right now that we can't overcome right now. In the future we're planning to migrate to WebGPU, but most likely it won't happen this year.

Memory-wise, you don't need to have a lot. But it's nice to have a fast GPU. I can say that that all M1-M3 macs work pretty well with Cosmos.
